tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/arrays/ArraySizeDefinitionTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+*  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.ballerinalang.test.statements.arrays;
+
+import org.ballerinalang.test.util.BAssertUtil;
+import org.ballerinalang.test.util.BCompileUtil;
+import org.ballerinalang.test.util.CompileResult;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+/**
+ * Array size definition test.
+ *
+ */
+public class ArraySizeDefinitionTest {
+
+    private String sizeMismatchError = "size mismatch in sealed array. expected '2', but found '3'";
+    private String invalidReferenceExpressionError = "invalid reference expression " +
+            "'intLength' as array size: expected a constant reference expression";
+    private String incompatibleTypeError = "incompatible types: expected 'int', found 'string'";
+    private String undefinedSymbolError = "undefined symbol 'length'";
+
+    @Test(groups = { "disableOnOldParser" })
+    public void testCompilationSizeReferenceErrors() {
+        int index = 0;
+        CompileResult result = BCompileUtil.compile(
+                "test-src/statements/arrays/array-size-test.bal");
+        Assert.assertEquals(result.getDiagnostics().length, 20);
```

Comment:
Let's move this to the end and assert equality between `index` and `result.getDiagnostics().length`, that way we also make sure we've asserted all the errors.